test/hotspot/jtreg/runtime/cds/appcds/DumpClassList.java

```diff
@@ -74,26 +75,33 @@ public static void main(String[] args) throws Exception {
         String appendJar = JarBuilder.build("bootappend", "boot/append/Foo");
 
         // dump class list
-        ProcessBuilder pb = ProcessTools.createTestJvm(
-            "-XX:DumpLoadedClassList=" + classList,
-            "--patch-module=java.base=" + patchJar,
-            "-Xbootclasspath/a:" + appendJar,
-            "-cp",
-            appJar,
-            appClass[0]);
-        OutputAnalyzer output = TestCommon.executeAndLog(pb, "dumpClassList");
-        TestCommon.checkExecReturn(output, 0, true,
-                                   "hello world",
-                                   "skip writing class java/lang/NewClass") // skip classes outside of jrt image
-            .shouldNotContain("skip writing class boot/append/Foo");        // but classes on -Xbootclasspath/a should not be skipped
+        CDSOptions opts = (new CDSOptions())
+            .setUseVersion(false)
+            .setXShareMode("auto")
+            .addPrefix("-XX:DumpLoadedClassList=" + classList,
+                       "--patch-module=java.base=" + patchJar,
+                       "-Xbootclasspath/a:" + appendJar,
+                       "-cp",
+                       appJar,
+                       appClass[0]);
+        CDSTestUtils.run(opts)
+            .assertNormalExit(output -> {
+                output.shouldContain("hello world");
+                // skip classes outside of jrt image
+                output.shouldContain("skip writing class java/lang/NewClass");
+                // but classes on -Xbootclasspath/a should not be skipped
+                output.shouldNotContain("skip writing class boot/append/Foo");
+            });
 
-        output = TestCommon.createArchive(appJar, appClass,
-                                          "-Xbootclasspath/a:" + appendJar,
-                                          "-Xlog:class+load",
-                                          "-XX:SharedClassListFile=" + classList);
-        TestCommon.checkDump(output)
+        opts = (new CDSOptions())
+            .setClassList(appClass)
+            .addPrefix("-cp", appJar,
+                       "-Xbootclasspath/a:" + appendJar,
+                       "-Xlog:class+load",
+                       "-XX:SharedClassListFile=" + classList);
+        OutputAnalyzer out = CDSTestUtils.createArchive(opts);
+        CDSTestUtils.checkDump(out, "[info][class,load] boot.append.Foo")
             .shouldNotContain("Preload Warning: Cannot find java/lang/invoke/LambdaForm")
-            .shouldNotContain("Preload Warning: Cannot find boot/append/Foo")
-            .shouldContain("[info][class,load] boot.append.Foo");
+            .shouldNotContain("Preload Warning: Cannot find boot/append/Foo");
```

Comment:
I think can be further simplified as

```
CDSTestUtils.createArchiveAndCheck(opts)
   .shouldNotContain("Preload Warning: Cannot find java/lang/invoke/LambdaForm")
   .shouldNotContain("Preload Warning: Cannot find boot/append/Foo")
   .shouldContain("[info][class,load] boot.append.Foo");
```

Probably most `createArchive()` calls can be replaced `with createArchiveAndCheck()`